examples/optsCommands.c

```diff
@@ -0,0 +1,161 @@
+			      "args", "remaining", KEY_END),
+		      KS_END);
+}
+
+int main (int argc, const char ** argv)
+{
+	KeySet * ks = createSpec ();
+	Key * errorKey = keyNew (BASE_KEY, KEY_END);
+
+	int result = elektraGetOpts (ks, argc, argv, (const char **) environ, errorKey);
+	if (result == -1)
+	{
+		fprintf (stderr, "ERROR: %s\n", keyString (keyGetMeta (errorKey, "error/reason")));
+		keyDel (errorKey);
+		ksDel (ks);
+		return EXIT_FAILURE;
+	}
+
+	if (result == 1)
+	{
+		char * help = elektraGetOptsHelpMessage (errorKey, NULL, NULL);
+		fprintf (stderr, "%s\n", help);
+		elektraFree (help);
+		keyDel (errorKey);
+		ksDel (ks);
+		return EXIT_SUCCESS;
+	}
+
+	printf ("A real implementation would now \n");
+
+	Key * lookup = ksLookupByName (ks, BASE_KEY "/printversion", 0);
+	if (lookup != NULL && elektraStrCmp (keyString (lookup), "1") == 0)
+	{
+		printf ("print version information\n");
+
+		keyDel (errorKey);
+		ksDel (ks);
+
+		return EXIT_SUCCESS;
+	}
+
+	lookup = ksLookupByName (ks, BASE_KEY "", 0);
+	if (lookup != NULL && elektraStrCmp (keyString (lookup), "getter") == 0)
+	{
+		lookup = ksLookupByName (ks, BASE_KEY "/getter/keyname", 0);
+		if (lookup == NULL || strlen (keyString (lookup)) == 0)
+		{
+			printf ("report the error 'missing parameter: keyname'\n");
+			keyDel (errorKey);
+			ksDel (ks);
+
+			return EXIT_SUCCESS;
+		}
+		printf ("read the key '%s'\n", keyString (lookup));
+
+		lookup = ksLookupByName (ks, BASE_KEY "/getter/verbose", 0);
+		if (lookup != NULL && elektraStrCmp (keyString (lookup), "1") == 0)
+		{
+			printf ("print where the read key value comes from\n");
+		}
+	}
+	else if (lookup != NULL && elektraStrCmp (keyString (lookup), "setter") == 0)
+	{
+		const char * keyname;
+
+		lookup = ksLookupByName (ks, BASE_KEY "/setter/keyname", 0);
+		if (lookup == NULL || strlen (keyname = keyString (lookup)) == 0)
+		{
+			printf ("report the error 'missing parameter: keyname'\n");
+			keyDel (errorKey);
+			ksDel (ks);
+
+			return EXIT_SUCCESS;
+		}
+		lookup = ksLookupByName (ks, BASE_KEY "/setter/value", 0);
+		if (lookup == NULL)
+		{
+			printf ("report the error 'missing parameter: value'\n");
+			keyDel (errorKey);
+			ksDel (ks);
+
+			return EXIT_SUCCESS;
+		}
+		printf ("write the value '%s' to the key '%s'\n", keyString (lookup), keyname);
+
+		lookup = ksLookupByName (ks, BASE_KEY "/setter/verbose", 0);
+		if (lookup != NULL && elektraStrCmp (keyString (lookup), "1") == 0)
+		{
+			printf ("print where the key value is stored now\n");
+		}
+	}
+	else
+	{
+		Key * arrayParent = ksLookupByName (ks, BASE_KEY "/dynamic", 0);
+		KeySet * dynamicCommand = elektraArrayGet (arrayParent, ks);
+
+		if (ksGetSize (dynamicCommand) > 0)
+		{
+			printf ("dynamically invoke the command '");
+			ksRewind (dynamicCommand);
+			Key * cur = NULL;
+			while ((cur = ksNext (dynamicCommand)) != NULL)
+			{
+				printf ("%s ", keyString (cur));
```

Comment:
I get here the output `dynamically invoke the command 'dynamic abc '` but would not expect `dynamic` to be executed`(but only abc)

Reply:
What was the shell command you invoked?

Reply:
```sh
bin/optsCommands dynamic abc
#> A real implementation would now 
#> dynamically invoke the command 'dynamic abc '

Reply:
Ah ok, I think your confusion stems from the use of `dynamically`  and `dynamic`.
It might become clearer, when you look at `bin/optsCommands test abc`:

```
bin/optsCommands test abc
#> A real implementation would now 
#> dynamically invoke the command 'test abc '
```

Reply:
Ahh, you are right. So you basically already reimplemented what `kdb` does :smile:  The output `dynamically invoke the command 'test' with the arguments 'abc '` would be imho more pretty, though.

Reply:
> The output dynamically invoke the command 'test' with the arguments 'abc ' would be imho more pretty, though.

I will change the output.